scipy/stats/_hypotests.py

```diff
@@ -1433,3 +1435,242 @@ def cramervonmises_2samp(x, y, method='auto'):
+    """Result of `scipy.stats.tukeykramer`.
+
+    Attributes
+    ----------
+    sig_level : float
+            The desired significance level (copied from `tukeykramer` input).
+    statistic : float ndarray
+        The computed statistic of the test for each comparison. The
+        ``(i, j)`` index is the p-value for the ith - jth group comparison.
+    pvalue : float ndarray
+        The associated p-value from the studentized range distribution. The
+        ``(i, j)`` index is the p-value for the ith - jth group comparison.
+
+    Methods
+    -------
+    confidence_inteval() :
+        Compute the confidence interval for the significance level
+        specified in the test.
+
+    References
+    ----------
+    .. [1] NIST/SEMATECH e-Handbook of Statistical Methods, "7.4.7.1. Tukey's
+           Method."
+           https://www.itl.nist.gov/div898/handbook/prc/section4/prc471.htm,
+           28 November 2020.
+    """
+
+    def __init__(self, sig_level, statistic, pvalue, _nargs, _nsamples,
+                 _stand_err):
+        self.sig_level = sig_level
+        self.statistic = statistic
+        self.pvalue = pvalue
+        self._nargs = _nargs
+        self._nsamples = _nsamples
+        self._stand_err = _stand_err
+
+    def confidence_interval(self):
+        # determine the critical value of the studentized range using the
+        # appropriate significance level, number of treatments, and degrees
+        # of freedom as determined by the number of data less the number of
+        # treatments. ("Confidence limits for Tukey's method")[1]. Note that
+        # in the cases of unequal sample sizes there will be a criterion for
+        # each group comparison.
+        params = (1 - self.sig_level, self._nargs,
+                  self._nsamples - self._nargs)
+        srd = distributions.studentized_range.ppf(*params)
+        # also called maximum critical value, the tukey criterion is the
+        # studentized range critical value * the square root of mean square
+        # error over the sample size.
+        tukey_criterion = srd * self._stand_err
+        # the confidence levels are determined by the
+        # `mean_differences` +- `tukey_criterion`
+        upper_conf = self.statistic + tukey_criterion
+        lower_conf = self.statistic - tukey_criterion
+        return ConfidenceInterval(low=lower_conf, high=upper_conf)
+
+
+def _tukeykramer_iv(args, sig_level):
+    args = [np.asarray(arg) for arg in args]
+    for arg in args:
+        if arg.ndim != 1:
+            raise ValueError("Input samples must be one-dimensional")
+        if arg.size <= 1:
+            raise ValueError("Input sample size must be greater than one.")
+        if np.isinf(arg).any():
+            raise ValueError("Input samples must be finite.")
+    if not 0 < sig_level < 1:
+        raise ValueError("Significance level should be between 0 and 1.")
+    return args
+
+
+def tukeykramer(*args, sig_level=.05):
```

Comment:
One of the other refguide failures seems to be because this needs to be listed to the documentation in `__init__.py`

Reply:
I was required to change `pagel` to `page_trend_test` to be more descriptive.`tukey_hsd` would be probably be more recognizable, and if the acronym is understood, it's more descriptive. I could also see `tukey_posthoc`.

Reply:
I like `tukey_hsd`. I will write in the description that it incorporates the tukey kramer method for unequal sample sizes.